src/Corax/Queries/UnaryMatch.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using Voron;
+
+namespace Corax.Queries
+{
+    public enum UnaryMatchOperation
+    {
+        GreaterThan,
+        GreaterThanOrEqual,
+        LessThan,
+        LessThanOrEqual,
+        Equal,
+        NotEqual,
+        Not
```

Comment:
Those aren't supported?

Reply:
`Not` and `NotEqual` haven't implemented them yet. Actually `Equal` already exist, different method, should be deleted.

Reply:
The reason this is interesting is that `negation` is an interesting problem to have. Lucene does that with exclusion on the other query clause.

Reply:
I would not know how to implement `Not` here. I pushed `Equals` and `NotEquals`